stdlib/email/src/main/java/org/ballerinalang/stdlib/email/client/PopClient.java

```diff
@@ -0,0 +1,132 @@
+ * Contains functionality of POP Client.
+ */
+public class PopClient {
+
+    private static final Logger log = LoggerFactory.getLogger(PopClient.class);
+
+    private PopClient() {
+        // A Singleton class.
+    }
+
+    /**
+     * Initializes the ObjectValue object with the POP Properties.
+     * @param clientEndpoint Represents the POP Client class
+     * @param config Properties required to configure the POP Session
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @return If an error occurs in the POP client, error
+     */
+    public static Object initClientEndpoint(ObjectValue clientEndpoint, MapValue<Object, Object> config,
+                                            boolean isPop) {
+        if (log.isDebugEnabled()) {
+            log.debug("[PopClient][InitClient] Calling getProperties");
+        }
+        Properties properties = PopUtil.getProperties(config, isPop);
+        Session session = Session.getInstance(properties, null);
+        try {
+            Store store;
+            if (isPop) {
+                store = session.getStore(PopConstants.POP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(PopConstants.PROPS_POP_HOST));
+            } else {
+                store = session.getStore(ImapConstants.IMAP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(ImapConstants.PROPS_IMAP_HOST));
+            }
+            clientEndpoint.addNativeData(PopConstants.PROPS_STORE, store);
+            clientEndpoint.addNativeData(PopConstants.PROPS_USERNAME,
+                    properties.getProperty(PopConstants.PROPS_USERNAME));
+            clientEndpoint.addNativeData(PopConstants.PROPS_PASSWORD,
+                    properties.getProperty(PopConstants.PROPS_PASSWORD));
+            return null;
+        } catch (NoSuchProviderException e) {
+            log.error("Failed initialize client properties : ", e);
+            return BValueCreator.createErrorValue(StringUtils.fromString(PopConstants.POP_ERROR_CODE),
+                    e.getMessage());
+        }
+    }
+
+    /**
+     * Read emails from the server.
+     * @param clientConnector Represents the POP Client class
+     * @param filter Criteria, which is used to read emails
+     * @return If successful return the received email, otherwise an error
+     */
+    public static Object readMessage(ObjectValue clientConnector, MapValue<Object, Object> filter) {
+        try {
+            Store store = (Store) clientConnector.getNativeData(PopConstants.PROPS_STORE);
+            String host = (String) clientConnector.getNativeData(PopConstants.PROPS_HOST);
+            String username = (String) clientConnector.getNativeData(PopConstants.PROPS_USERNAME);
+            String password = (String) clientConnector.getNativeData(PopConstants.PROPS_PASSWORD);
```

Comment:
Shall we keep this password as a byte[] and convert to string on demand at least? @ldclakmal  wdyt?

Reply:
Let's consider the update in next phase of development.

Reply:
@wggihan Yes, that makes these credentials more secure than keeping those just as string.